src/Servers/Kestrel/Core/src/ListenOptions.cs

```diff
@@ -84,6 +84,19 @@ internal ListenOptions(ulong fileHandle, FileHandleType handleType)
         /// <remarks>Defaults to HTTP/1.x and HTTP/2.</remarks>
         public HttpProtocols Protocols { get; set; } = DefaultHttpProtocols;
 
+        /// <summary>
+        /// Gets or sets a value that controls whether to the "Alt-Svc" header is included with the response.
+        /// The "Alt-Svc" header is used by clients to upgrade HTTP/1.1 and HTTP/2 connections to HTTP/3.
+        /// <para>
+        /// The "Alt-Svc" header is only included with a response if <see cref="Protocols"/> has either
+        /// HTTP/1.1 or HTTP/2 enabled, and HTTP/3 is enabled.
+        /// </para>
+        /// </summary>
+        /// <remarks>
+        /// Defaults to true.
+        /// </remarks>
+        public bool EnableAltSvc { get; set; } = true;
```

Comment:
Future proofing: if/when we do the HTTP/2 altsvc frame would it be controlled by the same flag, and would we not send the header for those HTTP/2 responses? Or would we do both?

Reply:
https://datatracker.ietf.org/doc/html/rfc7838#section-4
>    Note that it would be unwise to mix the use of Alt-Svc header fields
   with the use of ALTSVC frames, as the sequence of receipt might be
   hard to predict.